com.unity.ads.ios-support/Runtime/Plugins/iOS/SkAdNetworkManager.m

```diff
@@ -32,23 +32,43 @@ + (SkAdNetworkManager *)sharedInstance {
 }
 
 - (void)updateConversionValue:(NSInteger)conversionValue {
-    if (!self.isAvailable)
+    if (!self.isAvailable) {
+        NSLog(@"updateConversionValue::Not properly initialized");
         return;
+    }
 
-    SEL requestSelector = NSSelectorFromString(@"updateConversionValue:");
-    if ([self.SkAdNetworkClass respondsToSelector:requestSelector]) {
-        NSNumber *val = [NSNumber numberWithInteger:conversionValue];
-        [self.SkAdNetworkClass performSelector:requestSelector withObject:val];
+    id target = self.SkAdNetworkClass;
```

Comment:
Though its a proper implementation, I think for the future we should use this from UnitAds or from the shared tools from used by UnityAds.
We already have there a proper way of using reflection there and loading the framework


For example(not related to this change): the code that checks it the current device is a sim has already a potential error when we use M1.

Reply:
I cannot comment on this as I don't have full context. In GameGrowth, users do not install the UnityAds SDK and use either Unity or IronSource mediation to monetize. This package seemed useful for us and other Unity users to access the native conversion value API from C#, and it seemed coherent with the rest of the functionality it offers.

Reply:
What I mean here, UnityAds or I believe some other teams most likely write the same code. I just suggested to reuse it. And it doesnt mean we need to install UnityAds SDK. @UnitySteven.